src/main/java/org/example/MainServlet.java

```diff
@@ -1,19 +1,47 @@
 package org.example;
 
-import jakarta.servlet.annotation.WebServlet;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+import org.example.model.UrlPattern;
+import org.example.service.HelloService;
+import org.example.service.JsonResponseService;
+import org.example.service.TableService;
+
 import java.io.IOException;
-import java.io.PrintWriter;
 
 public class MainServlet extends HttpServlet {
 
+    private final HelloService helloService = HelloService.getInstance();
+    private final TableService tableService = TableService.getInstance();
+    private final JsonResponseService jsonResponseService = JsonResponseService.getInstance();
+
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
-        PrintWriter printWriter = resp.getWriter();
+        logRequest(req);
+
+        String requestURI = req.getRequestURI();
+        if (requestURI.equals(UrlPattern.HTML.getUrl())) {
+            resp.setContentType("text/html");
+            String tableResponse = tableService.generateHtmlTable();
+            resp.getWriter().write(tableResponse);
+        } else if (requestURI.equals(UrlPattern.HELLO.getUrl())) {
+            resp.setContentType("text/html");
+            String helloResponse = helloService.getHelloResponse();
+            resp.getWriter().write(helloResponse);
+        } else if (requestURI.equals(UrlPattern.JSON.getUrl())) {
+            resp.setContentType("application/json");
+            String jsonResponse = jsonResponseService.jsonSender();
+            resp.getWriter().write(jsonResponse);
+        }
```

Comment:
Send `not found` response, if user tries to access non-existing resource

Reply:
fix